js_modules/dagit/packages/core/src/runs/RunMetadataProvider.tsx

```diff
@@ -285,7 +280,12 @@ export function extractMetadataFromLogs(
         }
       }
 
-      if (log.__typename === 'ExecutionStepStartEvent') {
+      if (
+        log.__typename === 'ResourceInitStartedEvent' ||
+        log.__typename === 'StepWorkerStartingEvent'
+      ) {
+        upsertState(step, timestamp, IStepState.PREPARING);
```

Comment:
did you test in process execution? I think that would just skip `Preparing` state entirely which i think is fine behavior but worth double checking if you did not

Reply:
ResourceInitStartedEvent still fires in in process

Reply:
but only once for the entire in process run not associated with a specific step

Reply:
ah true... that was somewhat before though this PR I think? (in the case where it actually executes, we weren't showing a Preparing line at all since I don't believe there were any logs before the step startsed?)

I will be sure to test that nothing too crazy happens

Reply:
ya i highly suspect its fine / no change, but its worth including in your test plan in my opinion